### DISCUSSION TITLE
EDP PLUG EDP RE:DY

### DISCUSSION THREAD

**Discussion by freitas3001** [Koenkk/zigbee2mqtt#10553] [Devices]
Hi,

I have 3 EDP PLUG EDP RE:DY devices that I can't configure.
Someone can help me?

The first one is configured, but the values do not appear.
In the second and third, it appears as Unsupported.

Thanks 

![1](https://user-images.githubusercontent.com/15038736/147990754-960325ea-91db-469d-bf7e-48aa117a3404.png)
![ZB-SmartPlug-1 0 0 (1)](https://user-images.githubusercontent.com/15038736/147990785-f99b123c-1219-469b-8771-363992c7b5cc.jpg)
![ZB-SmartPlug-1 0 0 (2)](https://user-images.githubusercontent.com/15038736/147990795-566ad7f3-c996-4682-b2ac-9d04109b90ff.jpg)
![a1](https://user-images.githubusercontent.com/15038736/147990809-2f2420f1-424e-472b-9198-9ff34ac57cbe.png)
![a2](https://user-images.githubusercontent.com/15038736/147990813-adee7586-cf5c-4610-8a7a-89e7b184de64.png)
![a3](https://user-images.githubusercontent.com/15038736/147990815-daa7d78a-3841-4d5e-820f-5ae605b96863.png)
![a4](https://user-images.githubusercontent.com/15038736/147990818-712bfcb0-9049-495b-ba38-63e0e57696b4.png)
![a5](https://user-images.githubusercontent.com/15038736/147990822-464d15b9-3375-4285-a48b-9c39daad197d.png)
![a6](https://user-images.githubusercontent.com/15038736/147990827-b4b4d707-92a4-4a51-8c63-867c81682768.png)
![a7](https://user-images.githubusercontent.com/15038736/147990833-94c14147-d48b-4269-9770-74e2b15daf3c.png)
![a8](https://user-images.githubusercontent.com/15038736/147990836-b296894f-0219-43ad-809a-984198dcaed7.png)
![ZB-SmartPlugIR-1 0 0 (1)](https://user-images.githubusercontent.com/15038736/147990874-2f5412f4-3559-448a-b554-2ea70d8c83a8.jpg)
![ZB-SmartPlugIR-1 0 0 (2)](https://user-images.githubusercontent.com/15038736/147990877-06cacb94-f03b-4ee3-9bf6-9d1bffed0655.jpg)
![b1](https://user-images.githubusercontent.com/15038736/147990883-1b68331c-02e4-4396-a221-90f38498a681.png)
![b2](https://user-images.githubusercontent.com/15038736/147990886-f67fc18e-f032-4ca5-9f7d-e0476e860345.png)
![b3](https://user-images.githubusercontent.com/15038736/147990891-7a34c7af-ceca-47d7-9eeb-74b749196599.png)
![ZB-SmartSolarPlug-1 1 1 (1)](https://user-images.githubusercontent.com/15038736/147990904-7d35ce0f-bd7c-4695-8745-82a5ecbd4697.jpg)
![ZB-SmartSolarPlug-1 1 1 (2)](https://user-images.githubusercontent.com/15038736/147990916-bfbe78c2-c57b-4aa5-a186-c26cb54a42b5.jpg)
![c1](https://user-images.githubusercontent.com/15038736/147990918-b9264bcb-81d2-4afc-9a5e-1c716e9a7859.png)

**Comment by dgomes** [comment]
create a file 'edpredy_ir.js' next to your configuration.yaml with the contents:

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

module.exports = [
    {
        zigbeeModel: ['ZB-SmartPlugIR-1.0.0'],
        model: 'PLUG EDP RE:DY',
        vendor: 'EDP',
        description: 're:dy plug',
        fromZigbee: [fz.on_off, fz.metering],
        toZigbee: [tz.on_off],
        configure: async (device, coordinatorEndpoint, logger) => {
            const endpoint = device.getEndpoint(85);
            await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'seMetering']);
            endpoint.saveClusterAttributeKeyValue('seMetering', {divisor: 1000, multiplier: 1});
            await reporting.onOff(endpoint);
            //await reporting.readMeteringMultiplierDivisor(endpoint);
            await reporting.instantaneousDemand(endpoint);
        },
        exposes: [e.switch(), e.power(), e.energy()],
    },
];
```

then in your configuration.yaml add the external converter:

```
external_converters:
  - edpredy_ir.js
```
